torch/__init__.py

```diff
@@ -333,29 +334,40 @@ def _load_global_deps() -> None:
 
     except OSError as err:
         # Can only happen for wheel with cuda libs as PYPI deps
-        # As PyTorch is not purelib, but nvidia-*-cu12 is
-        cuda_libs: dict[str, str] = {
-            "cublas": "libcublas.so.*[0-9]",
-            "cudnn": "libcudnn.so.*[0-9]",
-            "cuda_nvrtc": "libnvrtc.so.*[0-9]",
-            "cuda_runtime": "libcudart.so.*[0-9]",
-            "cuda_cupti": "libcupti.so.*[0-9]",
-            "cufft": "libcufft.so.*[0-9]",
-            "curand": "libcurand.so.*[0-9]",
+        # As PyTorch is not purelib, but nvidia-*-cu12 is.
+        # 2 patterns defined for each lib - first is specific, 
+        # second is generic backup.
+        cuda_maj = int(cuda_version.split(".")[0])
+        cuda_libs: Dict[str, str] = {
+            "cublas": [f"libcublas.so.{cuda_maj}*", "libcublas.so.[0-9]"],
```

Comment:
What if user installed 2 minor versions of Nvidia wheels? Shouldn't it always prefer newest version among all available?